File: Benchmarking/Toggle.c
// toggle_gpio_fast.c
// Fast GPIO toggle using bcm2835 in C

#include <bcm2835.h>
#include <stdio.h>

#define PIN RPI_GPIO_P1_11  // Physical pin 11 = BCM GPIO 17

int main() {
    if (!bcm2835_init()) {
        printf("Failed to initialize bcm2835.\n");
        return 1;
    }

    bcm2835_gpio_fsel(PIN, BCM2835_GPIO_FSEL_OUTP);  // Set as output

    while (1) {
        bcm2835_gpio_set(PIN);    // Set high
        bcm2835_gpio_clr(PIN);    // Set low
    }

    bcm2835_close();  // Not reached
    return 0;
}

File: Benchmarking/Toggle.py
# toggle_gpio_fast.py
# Rapid GPIO toggle test using RPi.GPIO (Python)
import RPi.GPIO as GPIO
import time

PIN = 17  # BCM GPIO 17

GPIO.setmode(GPIO.BCM)
GPIO.setup(PIN, GPIO.OUT)

try:
    while True:
        GPIO.output(PIN, GPIO.HIGH)
        GPIO.output(PIN, GPIO.LOW)
except KeyboardInterrupt:
    pass
finally:
    GPIO.cleanup()
